code/map/MapSelectionHandler.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Sandbox;
+
+namespace TTTReborn.Map
+{
+    public partial class MapSelectionHandler : BaseNetworkable
+    {
+        // Player Id (long) -> Map name (string)
+        [Net]
+        public IDictionary<long, string> PlayerIdMapVote { get; set; }
+
+        // Map name (string) -> Map image (string)
+        [Net, Change]
+        public IDictionary<string, string> MapImages { get; set; }
+
+        public int TotalRoundsPlayed = 0;
+
+        public async Task Load()
+        {
+            List<string> mapNames = await GetTTTMapNames();
+            List<string> mapImages = await GetTTTMapImages(mapNames);
+
+            for (int i = 0; i < mapNames.Count; ++i)
+            {
+                MapImages[mapNames[i]] = mapImages[i];
+            }
+        }
+
+        private static async Task<List<string>> GetTTTMapNames()
+        {
+            Package result = await Package.Fetch(Global.GameName, true);
+            return result.GameConfiguration.MapList;
+        }
+
+        private static async Task<List<string>> GetTTTMapImages(List<string> mapNames)
+        {
+            List<string> mapPanels = new();
+            for (int i = 0; i < mapNames.Count; ++i)
+            {
+                Package result = await Package.Fetch(mapNames[i], true);
+                mapPanels.Add(result.Thumb);
+            }
+            return mapPanels;
+        }
```

Comment:
Good for now, but later, I think partial loading would be cool. So stuff is rendered and usable but getting populated with the right image if it's loaded.

Reply:
Sounds reasonable 👍